spswarehouse/powerschool/powerschool.py

```diff
@@ -0,0 +1,437 @@
+import logging
+import time
+import os
+
+from ducttape.utils import (
+    get_most_recent_file_in_dir
+)
+
+from selenium.webdriver.chrome.webdriver import WebDriver
+from selenium.webdriver.support.ui import WebDriverWait, Select
+from selenium.webdriver.common.by import By
+from selenium.webdriver.support import expected_conditions as EC
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.common.exceptions import NoSuchElementException
+
+ADMIN_LOGIN_PAGE_PATH = 'admin/pw.html'
+ADMIN_HOME_PAGE_PATH = 'admin/home.html'
+ADMIN_URL_SCHEME = 'https://'
+STATE_REPORTS_PAGE_PATH = 'admin/reports/statereports.html?repType=state'
+REPORT_QUEUE_REPORTWORKS_PAGE_PATH = 'admin/reportqueue/prhome.html'
+REPORT_QUEUE_SYSTEM_PAGE_PATH = 'admin/reportqueue/home.html'
+
+def log_into_powerschool_admin(driver: WebDriver, 
+                               username: str, 
+                               password: str, 
+                               host: str):
+    """
+    Using the supplied Selenium webdriver, log into PowerSchool Admin
+    with the given username, password, and host.
+    
+    Arguments:
+        driver (webdriver.Chrome()): a Selenium-powered Chrome Window
+        username (str): the PowerSchool Admin username
+        password (str): the PowerSchool Admin password
+        host (str): the PowerSchool Admin login-page
+        
+    Returns:
+        None: Selenium will log you into PowerSchool Admin
+    """
+    logging.info("Create webpage URL for PowerSchool Admin")
+    host_full = host + "/" + ADMIN_LOGIN_PAGE_PATH
+    logging.info(f"The webpage url is: {host_full}")
+    
+    logging.info("Go to webpage URL for PowerSchool Admin")
+    driver.get(host_full)
+    
+    logging.info("Find the username field within the login page")
+    elem = WebDriverWait(driver, 30).until(EC.presence_of_element_located((By.ID, 'fieldUsername')))
+    
+    logging.info("Clear any pre-filled values within the username field")
+    elem.clear()
+    
+    logging.info("Type your PowerSchool username")
+    elem.send_keys(username)
+    
+    logging.info("Find the password field within the HTML page")
+    elem = driver.find_element(By.ID, 'fieldPassword')
+    
+    logging.info("Type your PowerSchool password")
+    elem.send_keys(password)
+    
+    logging.info("Press enter to submit your credentials and complete your login.")
+    elem.send_keys(Keys.RETURN)
+
+    try:
+        logging.info("Waiting for 'Start Page' element to be visible to confirm successful login")
+        elem = WebDriverWait(driver, 30).until(EC.visibility_of_element_located((By.XPATH, "//h1[text()='Start Page']")))
+
+        logging.info("Successful login confirmed!")
+    except:
+        raise Exception("Unable to confirm successful login to PowerSchool. Please check your credentials.")
+
+def get_current_domain(driver: WebDriver):
```

Comment:
I'm thinking this should be designated as an internal function (by putting an underscore at the beginning of the name)? While this is an illusion in Python, it is standard practice.  Unless you think people would use this directly?

Reply:
Done.